src/builder.rs

```diff
@@ -216,26 +248,77 @@ impl TransactionBuilder {
     }
 
     /// build a RingCtTransaction and associated secrets
-    pub fn build(self, rng: impl RngCore + CryptoRng) -> Result<DbcBuilder> {
+    pub fn build(self, rng: &mut (impl RngCore + CryptoRng)) -> Result<DbcBuilder> {
```

Comment:
You should be able to simply make `rng` mut.
```suggestion
    pub fn build(self, mut rng: impl RngCore + CryptoRng) -> Result<DbcBuilder> {
```

Then you can replace all uses of `&mut *rng` with `&mut rng`